ast/check.go

```diff
@@ -58,8 +58,16 @@ func (tc *typeChecker) CheckBody(env *TypeEnv, body Body) (*TypeEnv, Errors) {
 			tc.err(err)
 		}
 
+		refErrors := len(vis.errs) > 0
+
 		if err := tc.checkExpr(env, expr); err != nil {
-			tc.err(err)
+			// Suppress errors if another error occurred that is more
+			// actionable. In this case, if there is a ref error and the
+			// expression error is due to a nil type, it's likely caused by
+			// inability to infer a value's type referred to in the expr.
+			if !refErrors || !causedByNilType(err) {
```

Comment:
it might be worth having a CLI option to be verbose? probably not but just a thought